jykuo-love-shiritori/twp
# [frontend] Shop Route fix

There's a issue that the function associate to "user view shop" before is not complete and wrong.

### Changes
- user can now view product list in home page
- user can navigate to shop where the product is from
- the shop page now show the real things ( in ur db )
  - shop icon, shop desc, products count
  - click view shop to view product ( and click the product item to view detail
  - click coupon to view shop coupon

### Testing Approach (if unconventional)
```sh
docker compose up --build
```
- click ``more`` in the product in main page
- go to the shop by clicking ``view shop`` from the detail product
- check product and shop coupon in the shop
- maybe edit seller's things to see if it shown on the shop page. path is ``shop/{sellerName}``.

### Additional context
~~ill add ``add to cart`` function after this merge.~~
done, just in my stash stack.
